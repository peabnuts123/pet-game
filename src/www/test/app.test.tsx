import { h } from "preact";
import { render } from "@testing-library/preact";

import App from "@app/components/app";

describe('App', () => {
  test('Matches snapshot', () => {
    // Setup
    const { baseElement } = render(<App />);

    // Test / Assert
    expect(baseElement).toMatchSnapshot();
  });
});

