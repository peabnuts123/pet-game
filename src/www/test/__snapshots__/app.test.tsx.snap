// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matches snapshot 1`] = `
<body>
  <div>
    <div>
      <header
        class="header"
      >
        <div
          class="header__container container"
        >
          <a
            class="header__brand is-active"
            data-testid="header__brand"
            href="/"
          >
            Pet Game
          </a>
          <nav
            class="header__nav--desktop"
          >
            <a
              class="header__nav-item--desktop"
              href="/taking-tree"
            >
              The Taking Tree
            </a>
            <div
              class="header__nav-item--desktop"
            >
              <div
                class="spinner spinner--inverted"
              >
                <div
                  class="rect rect1"
                />
                <div
                  class="rect rect2"
                />
                <div
                  class="rect rect3"
                />
                <div
                  class="rect rect4"
                />
                <div
                  class="rect rect5"
                />
              </div>
            </div>
          </nav>
          <nav
            class="header__nav--mobile"
          >
            <div
              class="header__nav-item--mobile"
            >
              <button
                class="button"
              >
                <svg
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="3"
                    x2="21"
                    y1="12"
                    y2="12"
                  />
                  <line
                    x1="3"
                    x2="21"
                    y1="6"
                    y2="6"
                  />
                  <line
                    x1="3"
                    x2="21"
                    y1="18"
                    y2="18"
                  />
                </svg>
              </button>
            </div>
            <div
              class="header__nav--mobile__container"
            >
              <a
                class="header__nav--mobile__item is-active"
                href="/"
              >
                Home
              </a>
              <a
                class="header__nav--mobile__item"
                href="/taking-tree"
              >
                The Taking Tree
              </a>
              <div
                class="header__nav--mobile__item--divider"
              />
              <div
                class="header__nav--mobile__item"
              >
                <div
                  class="spinner spinner--inverted"
                >
                  <div
                    class="rect rect1"
                  />
                  <div
                    class="rect rect2"
                  />
                  <div
                    class="rect rect3"
                  />
                  <div
                    class="rect rect4"
                  />
                  <div
                    class="rect rect5"
                  />
                </div>
              </div>
            </div>
          </nav>
        </div>
      </header>
      <div
        class="container u-padding-top-lg"
      >
        <div
          class="home"
        >
          <h1
            class="home__title"
          >
            Welcome!
          </h1>
          <p
            class="home__description"
          >
            Welcome to Pet Game, a totally original game that is 
            <em>
              definitely not
            </em>
             a clone of a popular virtual pet website. As you can probably tell, the site is only in the early stages of development right now, but eventually you will be able to have lots of fun here. You will be able to adopt pets, play games with them, and explore!
          </p>
          <h2
            class="home__what-can-i-do-title"
          >
            What things can I do?
          </h2>
          <p>
            Right now there is not a lot you can do, but you can take a poke around, visit The Taking Tree, or view your own profile (including your inventory).
          </p>
          <div
            class="home__list"
          >
            <div
              class="home__list-item"
            >
              <div
                class="spinner"
              >
                <div
                  class="rect rect1"
                />
                <div
                  class="rect rect2"
                />
                <div
                  class="rect rect3"
                />
                <div
                  class="rect rect4"
                />
                <div
                  class="rect rect5"
                />
              </div>
            </div>
            <div
              class="home__list-item"
            >
              <a
                class=""
                href="/taking-tree"
              >
                Visit The Taking Tree
              </a>
            </div>
            <div
              class="home__list-item"
            >
              <a
                class=""
                href="/user-profile"
              >
                View your profile
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
